gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -865,4 +1016,23 @@ public static Builder builder() {
    * @throws StorageException upon failure
    */
   List<Boolean> delete(BlobId... blobIds);
+
+  /**
+   * Returns a {@link BlobRewriter} object for the provided {@code rewriteRequest}. If source and
+   * destination objects share the same location and storage class the source blob is copied with a
+   * single call of {@link BlobRewriter#copyChunk()}, regardless of the {@link
+   * RewriteRequest#maxBytesRewrittenPerCall} parameter. If source and destination have different
+   * location or storage class multiple RPC calls might be needed depending on blob's size.
+   * <p>
+   * Example usage of blob rewriter:
+   * <pre>    {@code BlobRewriter rewriter = service.rewriter(rewriteRequest);}
+   *    {@code while(!rewriter.isDone()) {
+   *       rewriter.copyChunk();
+   *   }}
+   * </pre>
+   *
+   * @throws StorageException upon failure
```

Comment:
I don't think it does now, but that would not be the case if we always send the first RPC in this request as I suggested above.